### PR TITLE
Add gradient clipping to the Policy Gradient agent

### PR DESCRIPTION
**Pull request by dvoram**
Without gradient clipping the PolicyGradient agent was unstable, occasionally resulting in NaN probabilities.

I added max_gradient_norm argument to the __init__ function of the PolicyGradient agent with the default value of 10.0. If set to None, gradient clipping is disabled.

The gradient clipping is implemented using [tf.clip_by_norm](https://www.tensorflow.org/api_docs/python/tf/clip_by_norm) operation.

### CODE REVIEW

**Comment by dvoram** [comment]
@googlebot I signed it!

**Comment by fzvinicius** [comment]
Thanks for the PR, @dvoram. A few remarks:

- Some parameter settings are expected to induce bad or Nan gradients. Which game and params did you use? 
- Note that grad clipping was not needed in the reference papers (e.g. RPG paper) which policy gradient was used. That being said, your contribution is welcome, but please set the default to not use grad clip.
- Please use [clip_by_global_norm](https://www.tensorflow.org/api_docs/python/tf/clip_by_global_norm) instead. More details on the paper [On the difficulty of training Recurrent Neural Networks
](https://arxiv.org/abs/1211.5063).

**Comment by lanctot** [comment]
@dvoram , did you see that last message?

**Comment by dvoram** [comment]
Thank you for the comments. Sorry for that delay, I missed the message.

>     * Some parameter settings are expected to induce bad or Nan gradients. Which game and params did you use?

I actually used my own environment implementing a variation of the [Guessing game](https://gym.openai.com/envs/GuessingGame-v0/). The NaNs were produced even with the default parameters, got worse with learning rates increased.

>     * Note that grad clipping was not needed in the reference papers (e.g. RPG paper) which policy gradient was used. That being said, your contribution is welcome, but please set the default to not use grad clip.

This definitely makes sense. I will set the parameter to None by default.

>     * Please use [clip_by_global_norm](https://www.tensorflow.org/api_docs/python/tf/clip_by_global_norm) instead. More details on the paper [On the difficulty of training Recurrent Neural Networks
>       ](https://arxiv.org/abs/1211.5063).

Ok.
